Aztec/Classes/Private/Libxml2/Data/ElementNode.swift

```diff
@@ -903,6 +949,21 @@ extension Libxml2 {
             }
         }
 
+        func replace(range targetRange: NSRange, withNodeNamed nodeName: String, withAttributes attributes:[Attribute]) {
+            guard let textNode = lowestTextNodeWrapping(targetRange) else {
+                return
+            }
+            let absoluteLocation = textNode.absoluteLocation()
+            let localRange = NSRange(location:targetRange.location - absoluteLocation, length: targetRange.length)
+            textNode.split(forRange: localRange)
+            let imgNode = ElementNode(name: nodeName, attributes: attributes, children: [])
+            guard let index = textNode.parent?.children.indexOf(textNode) else {
```

Comment:
I'd add another guard before this one that does:

```swift
guard let textNodeParent = textNode.parent else {
    ...
}
```

So that we can avoid the optionals below.